rfd/0067-database-access-aws-redis

```diff
@@ -0,0 +1,221 @@
+
+Teleport does NOT support Redis configured in this state, and users are recommended to migrate the database to use ACL/RBAC.<sup>[1]</sup>
+
+#### Auth - Redis with ACL
+[Redis ACL](https://redis.io/docs/manual/security/acl/) (also known as RBAC for ElatiCache Redis) is introduced in Redis 6. Both ElatiCache and MemoryDB provide APIs to manage database users and user groups for ACL.
+
+As with all other databases, the database users should be created by "database admins" (by users, not Teleport) to assign desired permissions.
+
+When an user runs `tsh db connect --db-user <database-user> <aws-rediaws-redis-server>`, database agent automatically sends `auth <database-user> <password>` to Redis server upon succesfull connection, to prevent user from entering the password manually.
+
+To provide better security than using a static password, Teleport must take control of these database users and apply periodic password rotation (more details on secret store in later section). Users can separate the database users used by Teleport vs other client applications.
+
+Here is a sample database service config to specify which database users to manage:
+```
+db_service:
+  enabled: "yes"
+
+  aws:
+  # Discovers ElastiCache/MemoryDB Redis databases.
+  - types: ["elasticache", "memorydb"]
+    regions: ["us-west-1"]
+    tags:
+      "vpc-id": "xyz"
+
+  # Manage (e.g. rotate password for) database users matching provided tags.
+  - types: ["elasticache-user", "memorydb-user"]
+    regions: ["us-west-1"]
+    tags:
+      "managed-by-teleport": "true"
+
+  # Statically list database users (as alternative to using tags matching).
+  users:
+  - arn: "arn:aws:elasticache:ca-central-1:123456789000:user:steve-teleport"
+
+```
+The database service creates one secret with random password per database user, rotates it frequently (e.g. 15 minutes), and updates the database user with new password value using AWS APIs.
+
+When database service tries to establish connection with Redis server, the password is retrieved from the secret store to perform `auth <database-user> <password>`. If the secret does not exist or expired, the connection is aborted.
+
+In HA mode, minimum one database service is required to manage the database users. If more than one services try to rotate the password at the same time, the secret store implementation guarantees only the first succeeds.
+
+### Secret store
+Using a password is undesirable but necessary in situation like this. For ElastiCache/MemoryDB Redis usage, it is natural to use a AWS native solution (aka [AWS Secrets Manager](https://aws.amazon.com/secrets-manager/)). Though other secret stores could be supported in the future. <sup>[2]<sup>
+
+Here is a sample config for specifying the secret store in database service.
+```
+db_service:
+  ...
+
+  secert_store:
+    type: "aws-secrets-manager"
+    key_prefix: "teleport/"
```

Comment:
Can this be a part of the database configuration? And can it be optional (default to something reasonable)?

Reply:
I think it makes sense per database config can define secret store on its own. Though how would we apply this to auto discovered the databases? We can `secert_store` per aws matcher section?

the path can be something like `teleport/database/user/elasticache/xxxx`. maybe we can enforce this similar to the label and get rid of the config?

Reply:
Updated and moved to `AWS Secrets Manager` section below with `kms_key_id`